src/dali_backend.cc

```diff
@@ -261,14 +261,18 @@ class DaliModelInstance : public ::triton::backend::BackendModelInstance {
       std::vector<IBufferDescr> buffers;
       buffers.reserve(input_buffer_count);
       for (uint32_t buffer_idx = 0; buffer_idx < input_buffer_count; ++buffer_idx) {
-        auto buffer = input.GetBuffer(buffer_idx, device_type_t::CPU, device_id_);
+        auto buffer = input.GetBuffer(buffer_idx, device_type_t::CPU, DetermineDeviceId());
         buffers.push_back(buffer);
       }
       ret.push_back({input.Meta(), std::move(buffers)});
     }
     return ret;
   }
 
+  int32_t DetermineDeviceId() {
```

Comment:
```suggestion
  int32_t GetDeviceId() {
```
or (better?)
```suggestion
  int32_t GetDaliDeviceId() {
```
Determine suggests it's done once and then the device id is know - also, the name suggests complex logic.